core-plugins/src/main/java/co/cask/hydrator/plugin/batch/action/HDFSFileDeleteAction.java

```diff
@@ -87,7 +80,17 @@ public boolean accept(Path path) {
 
   @Override
   public void configurePipeline(PipelineConfigurer pipelineConfigurer) {
+    if (config.fileRegex != null) {
+      filter = new PathFilter() {
+        private final Pattern pattern = Pattern.compile(config.fileRegex);
```

Comment:
just do Pattern.compile(config.fileRegex). You don't need everything else.  If it throws an exception, catch it and throw an IllegalArgumentException about the file regex being invalid with the exception message.